### PR TITLE
Python 2 and 3 cross compatible code

### PR DESCRIPTION
**Pull request by ajtritt**
deploy 2/3 compatible code and some other new features to main development branch. 

A few things:
- This uses the six code from #60 
- test.py now has some flags for running subsets of tests
- There are now base test classes for doing roundtrip tests i.e. write a container to disk, read it back in, and ensure that it is the same
- To run a roundtrip test, I had to add some getters to NWBFile. NWBFile has a getter for retrieving TimeSeries data (NWBFile.get_raw_timeseries, NWBFile.get_stimulus, NWBFile.get_stimulus_template). This is the final step read functionality.

### CODE REVIEW

**Comment by neuromusic** [comment]
haven't reviewed, but endorse the single-branch py2/3 dev strategy

File: src/form/backends/hdf5/h5tools.py [hunk 1]
```diff
@@ -129,8 +129,8 @@ def write_builder(self, **kwargs):
         set_attributes(self.__file, f_builder.attributes)
 
 def get_type(data):
-    if isinstance(data, str):
-        return special_dtype(vlen=str)
+    if isinstance(data, text_type) or isinstance(data, string_types):
```
Comment (nicain) [inline src/form/backends/hdf5/h5tools.py]: I think you could use isinstance(data, (text_type, string_types)) here.

File: src/form/data_utils.py [hunk 2]
```diff
@@ -1,7 +1,8 @@
 from collections import Iterable
 import numpy as np
 from abc import ABCMeta, abstractmethod
-
+from six import with_metaclass
+import six
```
Comment (nicain) [inline src/form/data_utils.py]: Is six actually needed in this file?  Might be able to kill this line.

File: src/pynwb/retinotopy.py [hunk 3]
```diff
@@ -5,7 +5,6 @@
 from . import register_class, CORE_NAMESPACE
 from .core import NWBContainer
 
-#@register_class('ImageRetinotopy', CORE_NAMESPACE)    # make sure to uncomment this after this class is implemented
 class aimage(NWBContainer):
```
Comment (nicain) [inline src/pynwb/retinotopy.py]: Using lower case seems non-pep8; maybe change this in another PR

**Review by nicain — DISMISSED** [review]
Big PR with lots of commits, but looks fine to merge!

**Comment by ajtritt** [comment]
@nicain thanks for the input. I addressed your comments

**Pull request merged**